Python/ceval.c

```diff
@@ -2148,6 +2148,7 @@ do_raise(PyThreadState *tstate, PyObject *exc, PyObject *cause)
                               "calling %R should have returned an instance of "
                               "BaseException, not %R",
                               cause, Py_TYPE(fixed_cause));
+                Py_DECREF(fixed_cause);
                 goto raise_error;
             }
             Py_DECREF(cause);
```

Comment:
Just checking so I'm clear: `PyException_SetCause` steals a reference right? That's why we're not decrefing `fixed_cause` here?

Reply:
AFAIR, yes. `PyException_Set*` always steal refs IIRC, but I will check this again (it's always a pain to remember which function steals or borrows)